evmbin/src/display/json.rs

```diff
@@ -59,25 +95,37 @@ impl Informant {
 	fn informant_trace(informant: &Informant, gas_used: U256) -> String {
 		let info = ::evm::Instruction::from_u8(informant.instruction).map(|i| i.info());
 
-		json!({
-			"pc": informant.pc,
-			"op": informant.instruction,
-			"opName": info.map(|i| i.name).unwrap_or(""),
-			"gas": format!("{:#x}", gas_used.saturating_add(informant.gas_cost)),
-			"gasCost": format!("{:#x}", informant.gas_cost),
-			"memory": format!("0x{}", informant.memory.to_hex()),
-			"stack": informant.stack,
-			"storage": informant.storage,
-			"depth": informant.depth,
-		}).to_string()
+		// Reference: https://serde.rs/attr-skip-serializing.html
+		let trace_data =
+			TraceData {
+				pc: informant.pc,
+				op: informant.instruction,
+				op_name: info.map(|i| i.name).unwrap_or(""),
+				gas: &format!("{:#x}", gas_used.saturating_add(informant.gas_cost)),
+				gas_cost: &format!("{:#x}", informant.gas_cost),
+				memory: &format!("0x{}", informant.memory.to_hex()),
+				stack: &informant.stack,
+				storage: &informant.storage,
+				depth: informant.depth,
+			}
+		;
+
+		serde_json::to_string(&trace_data).expect("serialization cannot fail; qed")
 	}
 }
 
 impl vm::Informant for Informant {
 	type Sink = ();
 
 	fn before_test(&mut self, name: &str, action: &str) {
-		println!("{}", json!({"action": action, "test": name}));
+		let init_message =
+			InitMessage {
+				action: &format!("{}", action),
+				test: &format!("{}", name),
+			}
+		;
+
+		println!("{:?}", init_message);
```

Comment:
Could this be a log instead, maybe `info!(…)`?

Reply:
We want the output to go to `stdout` and it should be in a raw form (we generate JSON)

Reply:
Actually, the current output is a bit different than it used to be: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e97605d28a4ab5e2757c2b17f85e6f61.
I don't know if it really matters though.

Reply:
@ordian Great catch. Yeah - it's wrong now, it has to be proper JSON as the name of the file suggests.

Reply:
@dvc94ch thanks, i just noticed we're using the env_logger crate https://docs.rs/env_logger/0.6.1/env_logger/ as the logging implementation https://docs.rs/log/0.4.6/log/#available-logging-implementations that we can use to do `info!(..)` 

@tomusdrw thanks, so it looks like we should use raw string literals https://doc.rust-lang.org/reference/tokens.html#raw-string-literals as used in this example, so it would become something like the following:
```
let var1 = "test1";
let json = format!(r#"{{"type": "type1", "type2": {}}}"#, var1);
println!("{}", json) // => {"type2": "type1", "type2": test1}
```

But then to use the env_logger crate so it goes to std_out (instead of std_err by default)
https://github.com/sebasmagri/env_logger#configuring-log-target we'd need to change it to something like:
```
use std::env;
use env_logger::{Builder, Target};

let mut builder = Builder::from_default_env();
builder.target(Target::Stdout);

builder.init();

... 

let raw_json = format!(r#"{:?}"#, init_message);
info!("{:?}", raw_json)
```

Reply:
@ltfschoen I think we should use something like this
```suggestion
		let s = serde_json::to_string(&init_message).expect("serialization cannot fail; qed");
		println!("{}", s);
```
here and below

Reply:
What do you think about adding a newline between each log and tab indenting the actual log dump under headings that correspond to the struct name. i.e.

```
info!("\nInit Message: \n\t{:?}", init_message);
```